Building with the Win11 Insider SDK installed fails the build

### Repro

1. Install the Windows 11 Beta Build: https://blogs.windows.com/windows-insider/2021/06/24/preparing-for-insider-preview-builds-of-windows-11/
2. Install the Windows 11 Insisder SDK: https://www.microsoft.com/en-us/software-download/windowsinsiderpreviewSDK
3. Clone the runtime repo: `git clone https://github.com/dotnet/runtime`
4. Install any other pre-requisites required to build the repo: https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/windows-requirements.md
5. Attempt to build the runtime: https://github.com/dotnet/runtime/blob/main/docs/workflow/building/coreclr/README.md

### Expected behavior

The repo should build succesfully

### Actual behavior

Several components fail to build due to various issues, noting that only the first error of each "kind" was listed below:

#### MASM : warning A4018:invalid command-line option : /guard:cf
```
[238/2390] Building ASM_MASM object vm\wks\CMakeFiles\cee_wks_core.dir\__\amd64\CrtHelpers.asm.obj
  Microsoft (R) Macro Assembler (x64) Version 14.29.30040.0
  Copyright (C) Microsoft Corporation.  All rights reserved.

  MASM : warning A4018:invalid command-line option : /guard:cf
```

#### ucrt\stdalign.h(21): warning C4005: 'alignas': macro redefinition
```
[42/466] Building C object pal\src\libunwind\src\CMakeFiles\libunwind_xdac.dir\mi\mempool.c.obj
  FAILED: pal/src/libunwind/src/CMakeFiles/libunwind_xdac.dir/mi/mempool.c.obj
  C:\PROGRA~2\MICROS~2\2019\COMMUN~1\VC\Tools\MSVC\1429~1.300\bin\Hostx86\x64\cl.exe  /nologo -DBUILDENV_DEBUG=1 -DCROSS_COMPILE -DDEBUG -DDISABLE_CONTRACTS -DHAVE_CONFIG_H=1 -DHAVE_UNW_GET_ACCESSORS -DHAVE___THREAD=0 -DHOST_64BIT -DHOST_AMD64 -DHOST_WINDOWS -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_STRING=\"\" -DTARGET_64BIT -DTARGET_AMD64 -DTARGET_LINUX -DTARGET_UNIX -DUNW_REMOTE_ONLY -DURTBLDENV_FRIENDLY=Debug -D_CRT_SECURE_NO_WARNINGS -D_DBG -D_DEBUG -D_GNU_SOURCE -D_Thread_local="" -D__amd64__ -D__linux__ -D__x86_64__ -Ipal\src\libunwind\src -IC:\Users\tagoo\Source\repos\runtime\src\coreclr\pal\src\libunwind\src -IC:\Users\tagoo\Source\repos\runtime\src\coreclr\pal\src\libunwind\include\tdep -IC:\Users\tagoo\Source\repos\runtime\src\coreclr\pal\src\libunwind\include -Ipal\src\libunwind\include\tdep -Ipal\src\libunwind\include -IC:\Users\tagoo\Source\repos\runtime\src\coreclr\pal\src\libunwind\include\win /DWIN32 /D_WINDOWS  /guard:cf /guard:ehcont /Zi /Ob0 /Od /RTC1 -MTd /Od /nologo /W3 /WX /Oi /Oy- /Gm- /Zp8 /Gy /GS /fp:precise /FC /MP /Zm200 /Zc:strictStrings /Zc:wchar_t /Zc:inline /Zc:forScope /wd4065 /wd4100 /wd4127 /wd4189 /wd4200 /wd4201 /wd4245 /wd4291 /wd4456 /wd4457 /wd4458 /wd4733 /wd4838 /wd4960 /wd4961 /wd5105 /we4007 /we4013 /we4102 /we4551 /we4700 /we4640 /we4806 /w34092 /w34121 /w34125 /w34130 /w34132 /w34212 /w34530 /w35038 /w44177 /Zi /ZH:SHA_256 /source-charset:utf-8 /homeparams /TC /permissive- -wd4068 -wd4146 -wd4244 -wd4267 -wd4334 -wd4311 -wd4475 -wd4477 -std:c11 /showIncludes /Fopal\src\libunwind\src\CMakeFiles\libunwind_xdac.dir\mi\mempool.c.obj /Fdpal\src\libunwind\src\CMakeFiles\libunwind_xdac.dir\ /FS -c C:\Users\tagoo\Source\repos\runtime\src\coreclr\pal\src\libunwind\src\mi\mempool.c
  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdalign.h(21): error C2220: the following warning is treated as an error
  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdalign.h(21): warning C4005: 'alignas': macro redefinition
  C:\Users\tagoo\Source\repos\runtime\artifacts\obj\coreclr\Linux.x64.Debug\crossgen\pal\src\libunwind\include\config.h(26): note: see previous definition of 'alignas'
```

#### cl : Command line warning D9025 : overriding '/TP' with '/TC'
```
[54/466] Building CXX object pal\src\libunwind\src\CMakeFiles\libunwind_xdac.dir\__\__\exception\remote-unwind.cpp.obj  cl : Command line warning D9025 : overriding '/TP' with '/TC'
  cl : Command line warning D9025 : overriding '/TC' with '/TP'
```

--- a/src/coreclr/CMakeLists.txt
+++ b/src/coreclr/CMakeLists.txt
@@ -1,8 +1,15 @@
 cmake_minimum_required(VERSION 3.6.2)
 
-cmake_policy(SET CMP0042 NEW)
+cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH is enabled by default.
+
+if (CMAKE_VERSION VERSION_GREATER 3.7 OR CMAKE_VERSION VERSION_EQUAL 3.7)
+  cmake_policy(SET CMP0066 NEW) # Honor per-config flags in try_compile() source-file signature.
+endif()
+if (CMAKE_VERSION VERSION_GREATER 3.8 OR CMAKE_VERSION VERSION_EQUAL 3.8)
+  cmake_policy(SET CMP0067 NEW) # Honor language standard in try_compile() source-file signature
+endif()
 if (CMAKE_VERSION VERSION_GREATER 3.15 OR CMAKE_VERSION VERSION_EQUAL 3.15)
-  cmake_policy(SET CMP0091 NEW)
+  cmake_policy(SET CMP0091 NEW) # MSVC runtime library flags are selected by an abstraction.
 endif()
 
 # Set the project name

